src/posix/client.c

```diff
@@ -67,6 +67,55 @@ static void InputCallback(char *aLine)
 }
 #endif // OPENTHREAD_USE_READLINE
 
+static bool ContainsCommandReply(const char *aBuf)
+{
+    return strstr(aBuf, "Done") != NULL || strstr(aBuf, "Error");
+}
+
+static void SendBlockingCommand(int aArgc, char *aArgv[])
+{
+    char   buffer[OPENTHREAD_CONFIG_DIAG_CMD_LINE_BUFFER_SIZE];
+    char   lineBuffer[OPENTHREAD_CONFIG_DIAG_CMD_LINE_BUFFER_SIZE];
+    size_t now = 0;
+
+    for (int i = 0; i < aArgc; i++)
+    {
+        otEXPECT_ACTION(write(sSessionFd, aArgv[i], strlen(aArgv[i])) >= 0, perror("Failed to send command"));
+        otEXPECT_ACTION(write(sSessionFd, " ", 1) >= 0, perror("Failed to send command"));
+    }
+    otEXPECT_ACTION(write(sSessionFd, "\n", 1) >= 0, perror("Failed to send command"));
+
+    while (true)
+    {
+        ssize_t rval = read(sSessionFd, buffer, sizeof(buffer));
+
+        otEXPECT(rval >= 0);
+        for (ssize_t i = 0; i < rval; i++)
+        {
+            lineBuffer[now++] = buffer[i];
+
+            if (buffer[i] == '\r' || buffer[i] == '\n' || now >= sizeof(lineBuffer))
+            {
+                lineBuffer[now] = '\0';
+                if (ContainsCommandReply(lineBuffer))
+                {
+                    break;
+                }
+                now = 0;
+            }
+        }
+        lineBuffer[now] = '\0';
+        write(STDOUT_FILENO, buffer, rval);
```

Comment:
How about moving this `write` call to above the `for-loop` and exit immediately rather than breaking once `Done` or `Error` is detected? This should help remove redundant calls to `ContainsCommandReply`?